apps/src/makerlab/BoardController.js

```diff
@@ -28,207 +30,210 @@ const J5_CONSTANTS = {
-    Tap: PlaygroundIO.Tap,
-    Accelerometer: five.Accelerometer,
-    TouchSensor: TouchSensor
-  };
-
-  Object.keys(componentConstructors).forEach(function (key) {
-    codegen.customMarshalObjectList.push({instance: componentConstructors[key]});
-    jsInterpreter.createGlobalProperty(key, componentConstructors[key]);
-  });
-
-  Object.keys(this.prewiredComponents).forEach(function (key) {
-    jsInterpreter.createGlobalProperty(key, this.prewiredComponents[key]);
-  }.bind(this));
-};
+  installComponentsOnInterpreter(codegen, jsInterpreter) {
+    /**
+     * Set of classes used by interpreter to understand the type of instantiated
+     * objects, allowing it to make methods and properties of instances available.
+     */
+    const componentConstructors = {
+      Led: five.Led,
+      Board: five.Board,
+      RGB: five.Led.RGB,
+      Button: five.Button,
+      Switch: five.Switch,
+      Piezo: five.Piezo,
+      Thermometer: five.Thermometer,
+      Sensor: five.Sensor,
+      Pin: five.Pin,
+      CapTouch: PlaygroundIO.CapTouch,
+      Tap: PlaygroundIO.Tap,
+      Accelerometer: five.Accelerometer,
+      TouchSensor: TouchSensor
+    };
+
+    Object.keys(componentConstructors).forEach(function (key) {
+      codegen.customMarshalObjectList.push({instance: componentConstructors[key]});
+      jsInterpreter.createGlobalProperty(key, componentConstructors[key]);
+    });
 
-BoardController.prototype.reset = function () {
-  if (!this.board_) {
-    return;
+    Object.keys(this.prewiredComponents).forEach(function (key) {
+      jsInterpreter.createGlobalProperty(key, this.prewiredComponents[key]);
+    }.bind(this));
   }
 
-  const resetComponent = (component) => {
-    try {
-      if (component.state && component.state.intervalId) {
-        clearInterval(component.state.intervalId);
-      } else if (component.state && component.state.interval) {
-        clearInterval(component.state.interval);
-      }
-      if (component.stop) {
-        component.stop();
-      }
-      if (component.off) {
-        component.off();
-      }
-      if (component.threshold) {
-        // Reset sensor thresholds to 1.0 in case changed.
-        component.threshold = 1;
-      }
-      if (component.removeAllListeners) {
-        component.removeAllListeners();
-      }
-    } catch (error) {
-      console.log('Error trying to cleanup component', error);
-      console.log(component);
+  reset() {
+    if (!this.board_) {
+      return;
     }
-  };
-
-  // Components which do not get registered with the johnny-five board, but
-  // which can be reset in the same way.
-  var standaloneComponents = [
-    this.prewiredComponents.tap,
-    this.prewiredComponents.touch
-  ];
-  this.board_.register.concat(standaloneComponents).forEach(resetComponent);
-};
 
-BoardController.prototype.pinMode = function (pin, modeConstant) {
-  this.board_.pinMode(pin, modeConstant);
-};
+    const resetComponent = (component) => {
```

Comment:
Our styleguide isn't particular, but I'm actually not a fan of an arrow function in this case; I've been trying to use them only when I want the function to be anonymous, and the pattern here effectively names it.